joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -270,7 +276,70 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
       RubyOperators.hashInitializer,
       DispatchTypes.STATIC_DISPATCH
     )
-    callAst(call, argumentAsts)
+    scope.popScope()
+    blockAst(block, List(callAst(call, argumentAsts)))
+  }
+
+  protected def astForAssociationHash(node: Association, tmp: String): List[Ast] = {
+    node.key match {
+      case rangeExpr: RangeExpression =>
+        val expandedList = generateStaticLiteralsForRange(rangeExpr).map(x => {
```

Comment:
Just a style nit-pick, but you can make `.map(x => { ... })` into `.map { x => ... }`.